examples/solids/src/setup-libceed.c

```diff
@@ -734,6 +733,35 @@ PetscErrorCode SetupLibceedFineLevel(DM dm, DM dm_energy, DM dm_diagnostic,
   data[fine_level]->qf_diagnostic = qf_diagnostic;
   data[fine_level]->op_diagnostic = op_diagnostic;
 
+  // ---------------------------------------------------------------------------
+  // Enzyme-AD
+  // ---------------------------------------------------------------------------
+  // Create the QFunction and Operator that frees memory created for Enzyme-AD
+  // ---------------------------------------------------------------------------
+  if (problem_data.tape) {
+    CeedBasis basis_tape;
+    CeedOperator op_tape;
+    CeedQFunction qf_tape;
+    CeedInt idx = problem_data.number_fields_stored - 1, num_comp_tape = 1;
+    CeedBasisCreateTensorH1Lagrange(ceed, dim, num_comp_tape, P, Q,
```

Comment:
The tape is at the quadrature points, so the basis should be `CEED_BASIS_COLLOCATED` and an eval mode of `CEED_EVAL_NONE`

Reply:
```C

    CeedOperatorSetField(op_residual, problem_data.field_names[i],
                         data[fine_level]->elem_restr_stored_fields_i[i],
                         CEED_BASIS_COLLOCATED,
                         data[fine_level]->stored_fields[i]);
```
From `setup-libceed.c`

Reply:
Thanks! It does free it now.